### PR TITLE
Fix Interaction System doesn't follow the FSM

### PR DESCRIPTION
**Pull request by AlexandreGheraibia**
issue : https://github.com/UnityTechnologies/open-project-1/issues/342

**How did you resolve this issue?**  

In interactionManager, I divide the function OnInteractionButtonPress in two  OnInteractionButtonPress and OnInteractionResponseActivated. The last one is associated with a VoidEventChannel named InteractionResponse.

I also create an ActionScript and the associated action named ActiveInteractionResponse_OnEnter which raise the event InteractionResponse.

**How can it be verified that the issue has actually been resolved?**  

Try to pick up an object while you are jumping or attacking
Try to cook and see that you can't.

### CODE REVIEW

File: UOP1_Project/Assets/Scripts/Characters/StateMachine/Actions/ActiveInteractionResponse_OnEnterSO.cs [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+{
+	protected new ActiveInteractionResponse_OnEnterSO OriginSO => (ActiveInteractionResponse_OnEnterSO)base.OriginSO;
+	
+	public override void Awake(StateMachine stateMachine)
+	{
+	}
+	
+	public override void OnUpdate()
+	{
+		OriginSO.interactResponse.RaiseEvent();
```
Comment (DeivSky) [inline UOP1_Project/Assets/Scripts/Characters/StateMachine/Actions/ActiveInteractionResponse_OnEnterSO.cs:22]: This should probably be in `OnStateEnter`

Comment (AlexandreGheraibia) [reply UOP1_Project/Assets/Scripts/Characters/StateMachine/Actions/ActiveInteractionResponse_OnEnterSO.cs:22]: Yes, this Should be o_O', Thank you

**Comment by ciro-unity** [comment]
Hey @AlexandreGheraibia, I am merging this PR since I think the idea to break it into "before" and "after" is good. However, I am making some changes to clean up a lot of things (beyond this script).

I also think, if I'm not mistaken, that the PR as it is now breaks the cooking because Cooking is not a real state on the state machine, so nobody calls OnInteractionResponseActivated. But it's ok, I'm refactoring anyway! I'll explain more [on the forum](https://forum.unity.com/threads/world-interaction-system.1020709/unread).

**Comment by AlexandreGheraibia** [comment]
> 
> 
> Hey @AlexandreGheraibia, I am merging this PR since I think the idea to break it into "before" and "after" is good. However, I am making some changes to clean up a lot of things (beyond this script).
> 
> I also think, if I'm not mistaken, that the PR as it is now breaks the cooking because Cooking is not a real state on the state machine, so nobody calls OnInteractionResponseActivated. But it's ok, I'm refactoring anyway! I'll explain more [on the forum](https://forum.unity.com/threads/world-interaction-system.1020709/unread).

Yes, you got it 🥇

**Pull request merged**